lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1199,13 +1206,35 @@ class ConvertAtenLinearOp : public OpConversionPattern<AtenLinearOp> {
         loc, biasSizeCorrect,
         rewriter.getStringAttr("mismatching bias size for aten.linear"));
 
-    Value initTensor = rewriter.create<linalg::InitTensorOp>(
-        loc, ValueRange{inputDim0, weightDim0}, inputType.getElementType());
-    SmallVector<AffineMap> broadcastIndexingMaps = {
-        AffineMap::get(
-            /*dimCount=*/2, /*symbolCount=*/0, rewriter.getAffineDimExpr(1)),
-        rewriter.getMultiDimIdentityMap(2)};
-    SmallVector<StringRef> iteratorTypes(2, "parallel");
+    Value initTensor;
+    SmallVector<AffineMap> broadcastIndexingMaps;
+    Value transposedWeightInitTensor;
+    if (batchDim) {
+      initTensor = rewriter.create<linalg::InitTensorOp>(
+          loc, ValueRange{batchDim, inputDim0, weightDim0},
+          inputType.getElementType());
+      transposedWeightInitTensor = rewriter.create<linalg::InitTensorOp>(
+          loc, ValueRange{batchDim, weightDim1, weightDim0},
+          weightType.getElementType());
+      broadcastIndexingMaps = {
+          AffineMap::get(
+              /*dimCount=*/inputType.getRank(), /*symbolCount=*/0,
+              {rewriter.getAffineDimExpr(1 + restDim)}, context),
+          rewriter.getMultiDimIdentityMap(inputType.getRank())};
+    } else {
+      initTensor = rewriter.create<linalg::InitTensorOp>(
```

Comment:
Those 2 branches are very similar. It would be easier to read, if we use:
```
if (rank > 2) {
     initTensorShape = ..
     transposedWeightInitTensor = ...
     broadcastingAffineExpressions = ...
} else {

}
Create all the values here.
```
The differences between the 2 cases would be clearer this way.